server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java

```diff
@@ -739,11 +749,25 @@ private void checkifChopComplete(Set<StoredTabletFile> allFiles) {
     boolean completed;
 
     synchronized (this) {
+      if (closed) {
+        // if closed, do not attempt to transition to the MARKING state
+        return;
+      }
+      // when this returns true it means we transitioned to the MARKING state
       completed = fileMgr.finishChop(allFiles);
     }
 
     if (completed) {
-      markChopped();
+      try {
+        markChopped();
+      } finally {
+        synchronized (this) {
+          // transition the state from MARKING to NOT_ACTIVE
+          fileMgr.finishMarkingChop();
```

Comment:
Could you do something with the boolean that is returned from this method? It might not matter but since you are checking the state and returning an outcome, maybe a false should be handled.

Reply:
Could remove the boolean and change the method to have a precondition that expects the current chop state to be MARKING.  I made a suggested change for the method.  The way the code is currently written, it would be unexpected for the chop state to be anything else when this method is called.